app/com/linkedin/drelephant/spark/data/SparkComboApplicationData.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.drelephant.spark.data
+
+import scala.collection.JavaConverters
+import java.util.Properties
+
+import com.linkedin.drelephant.analysis.{ApplicationType, HadoopApplicationData}
+
+
+case class SparkComboApplicationData(
```

Comment:
SparkApplicationData already exists as an interface for the old fetcher to use: https://github.com/linkedin/dr-elephant/blob/dad905cdafa6aa7665059917a455cb601155fbd1/app/com/linkedin/drelephant/spark/data/SparkApplicationData.java. I didn't want to use this interface, so to preserve backwards compatibility for now I contrived SparkComboApplicationData.